[10.0][ADD] purchase_line_product_image:

kurniawanlucky opened the pull request:
* Show Product Image at purchase_order_line

kurniawanlucky committed:
Add purchase_line_product_image:
* Show Product Image at purchase_order_line

LoisRForgeFlow commented on purchase_line_product_image/__manifest__.py:
```diff
@@ -0,0 +1,17 @@
+{
```
Please can you add headers to all files following [OCA's template module](https://github.com/OCA/maintainer-tools/blob/master/template/module/__openerp__.py#L1-L3)

kurniawanlucky replied:
yes, it's done

kurniawanlucky committed:
Add
# -*- coding: utf-8 -*-
# © 2017
#   @Lucky Kurniawan <kurniawanluckyy@gmail.com>
# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
to headers

LoisRForgeFlow commented:
Thanks, can you please [check travis](https://travis-ci.org/OCA/purchase-workflow/jobs/294858479)? there are some lints failing.

kurniawanlucky committed:
add readme.rst

kurniawanlucky commented:
@Eficent Thanks for your help

kurniawanlucky committed:
change import

kurniawanlucky committed:
fix purchase_order_views.xml

kurniawanlucky committed:
fix some missing

LoisRForgeFlow reviewed (approved):
Tested on runbot. LGTM :+1: 

Thanks for the contribution!

lasley commented on purchase_line_product_image/README.rst:
```diff
@@ -0,0 +1,49 @@
+    :alt: License LGPL-3
+
+Purchase Order Line Product Image for purchases
+==========================================================
```
Please align the lenght of the `=` with the length of the header words (like below)

lasley commented on purchase_line_product_image/README.rst:
```diff
@@ -0,0 +1,49 @@
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues <https://github.com/OCA/purchase-workflow/issues>`_.
```
This is the old verbiage. Please update to align with [the template](https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst#bug-tracker)

lasley commented on purchase_line_product_image/__init__.py:
```diff
@@ -0,0 +1,6 @@
+# -*- coding: utf-8 -*-
+# © 2017
```
Your copyright should be one line. It's also worth noting that in a lot of countries, the symbol isn't a valid claim of copyright. I recommend using only the word - `Copyright 2017 My Company`

lasley commented on purchase_line_product_image/models/__init__.py:
```diff
@@ -0,0 +1,6 @@
+#   @Lucky Kurniawan <kurniawanluckyy@gmail.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from . import purchase_order
```
[Add a carriage return at the end of this line](https://robots.thoughtbot.com/no-newline-at-end-of-file)

lasley commented on purchase_line_product_image/models/purchase_order.py:
```diff
@@ -0,0 +1,16 @@
+
+    product_image = fields.Binary(
+        "image", related='product_id.product_tmpl_id.image_medium',
+        help="Small-sized image of the product. It is automatically "
```
All the attributes but `related` are superfluous and can be removed. Using `related` will bring over all the attributes from the original declaration, so you only need to declare attributes that are being changed 😉

kurniawanlucky replied:
hello @lasley, thanks for your review
can you help me give some example about "so you only need to declare attributes that are being changed"?
thanks

kurniawanlucky replied:
@lasley remove related and use "@api.onchange"?

kurniawanlucky replied:
yes, it's done

lasley reviewed (changes requested):
Thanks @kurniawanlucky - comment inline

kurniawanlucky committed:
purchase_line_product_image:
* fix some error and resize widget image

kurniawanlucky committed:
purchase_line_product_image:
* Fix views

kurniawanlucky committed:
purchase_line_product_image:
* fix product_image

kurniawanlucky committed:
purchase_line_product_image:
*fix manifest

kurniawanlucky committed:
purchase_line_product_image:
* fix py

kurniawanlucky committed:
[10.0][ADD] Purchase_order_barcodes:
* Add Product With Barcode at Purchase Order

kurniawanlucky committed:
[REMOVE]

lasley commented on purchase_line_product_image/README.rst:
```diff
@@ -0,0 +1,48 @@
+===========
+
+Bugs are tracked on `GitHub Issues
+<https://github.com/OCA/{project_repo}/issues>`_. In case of trouble, please
```
s/{project_repo}/purchase-workflow

kurniawanlucky replied:
yes, it's done
thanks @lasley

lasley reviewed (approved):
Thanks @kurniawanlucky - minor comment, but doesn't affect usability

kurniawanlucky committed:
purchase_line_product_image:
* fix readme

max3903 commented on purchase_line_product_image/README.rst:
```diff
@@ -0,0 +1,48 @@
+.. image:: https://img.shields.io/badge/licence-LGPL--3-blue.svg
+    :alt: License LGPL-3
```
@kurniawanlucky License says LGPL here but the files headers and manifest says AGPL.

kurniawanlucky replied:
@max3903 yes, it's done
thanks for your review

kurniawanlucky committed:
purchase_line_product_image:
* fix readme

lasley commented:
@max3903 - have your comments been attended to?
